packages/main/src/Menu.ts

```diff
@@ -542,9 +546,16 @@ class Menu extends UI5Element {
 		this._parentItemsStack.push(item);
 	}
 
+	_onfocusin(e: CustomEvent<any>): void {
+		const target = e.target as HTMLElement;
+		const menuListItem = target instanceof MenuListItem ? target : (target.getRootNode() as any).host;
```

Comment:
We never use `instanceof` in the framework because of multi-version scenarios. Use the duck-typing check (`hasAttribute("ui5-menu-item")`).

The `as any` cast is an anti-pattern and should only be used in extreme cases. Why not `as ShadowRoot` ?

Reply:
Sure, we could use the duck-typing check. We have to have in mind that the same won't work for the extended controls like `ui5-navigation-menu`. We'll be checking by attribute name, which won't be present there. For example we need to check if `target.hasAttribute("ui5-menu-li")` whereas in the `ui5-navigation-menu` we'll need to check for `target.hasAttribute("ui5-li")` duo to differences in the template. The good news are that this event isn't required there.

Reply:
Yeah, we don't enforce the Liskov substitution principle so we don't expect the descendant (ui5-navgation-menu-item) to have full compatibility with ui5-menu-item, so this should be fine.

Using `instanceof` is very risky, because if you have multiple runtimes of UI5 web components on the same page, and a different version managed to register `ui5-menu-item` on the window, then all "ui5-menu-item" tags will be upgraded with this version, and the `instanceof` check will fail and the code won't work.